grakn-graql/src/main/java/ai/grakn/graql/internal/query/DeleteQueryImpl.java

```diff
@@ -52,9 +46,8 @@ static DeleteQueryImpl of(Collection<? extends Var> vars, Match match) {
     }
 
     @Override
-    public Void execute() {
-        List<Answer> results = match().stream().collect(toList());
-        results.forEach(this::deleteResult);
+    public final Void execute() {
+        queryRunner().run(this);
```

Comment:
Out of curiosity is there no way we can make this a default method? Most of the `executes` seem to do this.

Reply:
I actually prefer this style, more of a preference thing since i still prefer keeping interfaces from implementations

Reply:
It would be nice to just have:

```
class AbstractQuery<T> {
    public final T execute() {
        queryRunner().run(this);
    }
}
```

Unfortunately, we can't do this, because Java does not know which `run` method to delegate to. In fact, it doesn't even know if the `Query<?>` class _has_ a `run` method on `QueryRunner`! For example, imagine if the user irritatingly decides to make their own implementation of `Query` and goes `query.execute()`.